runbotics/runbotics-orchestrator-ui/src/src-app/components/tables/Table/Table.types.ts

```diff
@@ -24,15 +26,48 @@ export interface DataTableProps<T extends object> {
     page?: number;
     loading?: boolean;
     subRowProperty?: string;
-    renderSubRow?: (row: Row<T>) => ReactNode;
+    renderSubRow?: (row: Row<T> | ProcessInstanceRow) => ReactNode;
     singleSelect?: boolean;
     autoHeight?: boolean;
 }
 
 export interface TableRowExpanderProps {
     row: Row;
+    handleClick?: (row: ProcessInstanceRow) => void;
+}
+
+export interface PanelInfoState {
+    show: boolean;
+    processInstanceId?: string;
+}
+
+export interface HistoryTableProps extends Omit<HTMLProps<HTMLDivElement>, 'title'> {
+    botId?: ProcessInstanceRequestCriteria['botId'];
+    processId?: ProcessInstanceRequestCriteria['processId'];
+    title?: ReactNode;
+    sx?: SxProps<Theme>;
+    rerunEnabled?: boolean;
 }
 
+export interface ProcessInstanceRow extends Omit<Row, 'subRows' | 'isExpanded'>{
+    original: InstanceExtendedWithSubProcesses;
+    subRows?: ProcessInstanceRow[];
+    isExpanded?: boolean;
+}
+
+
 export type Column<D extends object = {}> = ReactTableColumn<D> & {
     featureKeys?: FeatureKey[];
 };
+
+export interface getSubProcessesResponse { 
+    type: string;
+    payload: InstanceExtendedWithSubProcesses[];
+    meta: {
+        arg: {
+            processInstanceId: string;
+        };
+        requestId: string;
+        requestStatus: 'fulfilled';
+    }
+}
```

Comment:
this one does not belong to Table component. It's HistoryTable related. The same goes for all the new ones that you added here. Here should be only a generic type for the general-use Table component.

Reply:
done

Reply:
done